src/AbstractType.php

```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types=1);
+
+namespace Fortuneglobe\Types;
+
+use Fortuneglobe\Types\Interfaces\RepresentsScalarValue;
+
+/**
+ * Class AbstractType
+ * @package Fortuneglobe\Types
+ */
+abstract class AbstractType
+{
+	final public function __toString() : string
+	{
+		return $this->toString();
+	}
+
+	abstract public function toString() : string;
+
+	public function jsonSerialize() : string
```

Comment:
I cannot agree with this implementation. If I use a float value type, so it must serialize a float value und vice versa an unserialization must return a float. In the current state there's no ability to `unserialize( serialize( (string)0.7 ) )` and get the initial float value. So I highly recommend to use a getter for the value.